org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/VersionValidator.java

```diff
@@ -20,35 +20,42 @@
 
 public class VersionValidator {
 
-	private static Artifact setArtifact(DOMNode node) {
+	public static Artifact parseArtifact(DOMNode node) {
 		String groupId = "MissingGroupID";
 		String artifactId = "MissingArtifactID";
 		String version = "1.0.0";
 		String scope = "compile"; // Default scope if no scope is specified
 		String type = "jar"; // Default type is jar if no type is specified
 		String classifier = null; // Default classifier is null
-		for (DOMNode tag : node.getParentElement().getChildren()) {
-			switch (tag.getLocalName()) {
-			case "groupId":
-				groupId = tag.getChild(0).getNodeValue();
-				break;
-			case "artifactId":
-				artifactId = tag.getChild(0).getNodeValue();
-				break;
-			case "version":
-				version = tag.getChild(0).getNodeValue();
-				break;
-			case "scope":
-				scope = tag.getChild(0).getNodeValue();
-				break;
-			case "type":
-				type = tag.getChild(0).getNodeValue();
-				break;
-			case "classifier":
-				classifier = tag.getChild(0).getNodeValue();
-				break;
-			}
+		try {
```

Comment:
This try/catch was added as the parsing seemed to encounter NPE's. Will have to investigate further.